src/coreclr/nativeaot/System.Private.CoreLib/src/System/Type.NativeAot.cs

```diff
@@ -47,17 +41,29 @@ internal static unsafe RuntimeType GetTypeFromMethodTable(MethodTable* pMT)
         }
 
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private static unsafe RuntimeType GetTypeFromMethodTableSlow(MethodTable* pMT, ref UnsafeGCHandle handle)
+        private static unsafe RuntimeType GetTypeFromMethodTableSlow(MethodTable* pMT, ref RuntimeType? runtimeTypeCache)
         {
-            UnsafeGCHandle tempHandle = UnsafeGCHandle.Alloc(new RuntimeType(pMT));
+            // TODO: instead of fragmenting the frozen object heap, we should have our own allocator
```

Comment:
I have added check-box for this to https://github.com/dotnet/runtime/issues/91704